crates/sui-types/src/gas.rs

```diff
@@ -561,56 +561,52 @@ impl<'a> SuiGasStatus<'a> {
         }
     }
 
-    fn gas_used(&self) -> GasUnits {
+    pub fn gas_used(&self) -> GasUnits {
         let remaining_gas = self.gas_status.remaining_gas();
         self.init_budget
             .checked_sub(remaining_gas)
             .expect("Subtraction overflowed")
     }
 }
 
-/// Check whether the given gas_object and gas_budget is legit:
-/// 1. If the gas object has an address owner.
-/// 2. If it's enough to pay the flat minimum transaction fee
-/// 3. If it's less than the max gas budget allowed
-/// 4. If the gas_object actually has enough balance to pay for the budget
-/// 5. If total balance in gas object and extra input objects is sufficient
-/// to pay total amount of gas budget and extra amount to pay, extra input objects
-/// and extra amount to pay are only relevant in SUI payment transactions.
+// Check whether gas arguments are legit:
+// 1. Gas object has an address owner.
+// 2. Gas budget is between min and max budget allowed
 pub fn check_gas_balance(
     gas_object: &Object,
+    more_gas_objs: Vec<&Object>,
     gas_budget: u64,
     gas_price: u64,
-    more_gas_objs: Vec<Object>,
     cost_table: &SuiCostTable,
 ) -> UserInputResult {
+    // 1. Gas object has an address owner.
     if !(matches!(gas_object.owner, Owner::AddressOwner(_))) {
         return Err(UserInputError::GasObjectNotOwnedObject {
             owner: gas_object.owner,
         });
     }
 
+    // 2. Gas budget is between min and max budget allowed
```

Comment:
It's not very clear to me what these two checks do and are trying to prevent?

Reply:
you mean 1. and 2. I assume?
The first I am really not sure, I would have expected that to be true already (?)
the second checking that the number the user gave is in range (or so I read it and I hope it's doing that).
Are you asking whether the code is correct or why we do it?

Reply:
Yeah why does it need to be in a range?

Reply:
no idea, the min is important the max does not seem critical , not sure really. 
Funny enough I would have expected you or others here to have the answer.
@oxade @sblackshear @emmazzz anyone knows why we max the gas?

Reply:
Ok so we used to have max because gas_budget was in gas unit, and we need to make sure gas unit * internal transaction factor <= u64::MAX.

Reply:
got it! 
so you are telling me we do not need the max? 
or you think we should have a different max?
or you feel we should not have a max at all?
I don't think it particularly matters but that is also a reason to remove it if not needed

Reply:
also if the only reason for the ma was the overflow that does not seem an issue any longer for sure

Reply:
I guess if there is no other reason that the cost table needs a max, we don't need a max.
The moment we use the budget to multiply anything then we need a max (to make sure overflow cannot happen)

Reply:
So the question is do we still multiply it to convert to some internal units today?

Reply:
very good point and we may need a multiplier so it's good there for now.
Specifically we may still want to have an internal unit that is less than 1 mist, we'll see.
But I think this is good for now.
Thanks @lxfind for making me think of this! good stuff!